roles/example-cnf-app/tasks/app.yaml

```diff
@@ -91,30 +91,59 @@
   retries: 60
   delay: 5
   until:
-    - testpmd_pods.resources|length >= 1
+    - testpmd_pods.resources|length > 0
 
+# At least confirm that the first testpmd pod is up and running
 - name: check testpmd pod status to be running
   k8s_info:
     namespace: "{{ cnf_namespace }}"
     kind: Pod
     label_selectors:
       - example-cnf-type=cnf-app
   register: testpmd_pods
+  vars:
+    container_state_running_query: "resources[0].status.containerStatuses[?name=='testpmd'].state.running"
+    container_started_query: "resources[0].status.containerStatuses[?name=='testpmd'].started"
+    container_ready_query: "resources[0].status.containerStatuses[?name=='testpmd'].ready"
+    container_state_running: "{{ testpmd_pods | json_query(container_state_running_query) }}"
+    container_started: "{{ testpmd_pods | json_query(container_started_query) }}"
+    container_ready: "{{ testpmd_pods | json_query(container_ready_query) }}"
   retries: 60
   delay: 5
   until:
     - testpmd_pods.resources[0].status.phase == 'Running'
+    - container_state_running | length > 0
+    - container_started | length > 0
+    - container_started[0] | bool
+    - container_ready | length > 0
+    - container_ready[0] | bool
 
-- name: check cnfappmac resource (created by testpmd app)
+- name: check cnfappmac resource (created by testpmd app) count to be greater than 0
   k8s_info:
     namespace: "{{ cnf_namespace }}"
     kind: CNFAppMac
-  register: mac
+  register: cnfappmac
   retries: 120
   delay: 5
   until:
-    - mac.resources is defined
-    - mac.resources|length > 0
+    - cnfappmac.resources is defined
+    - cnfappmac.resources|length > 0
+
+# CNFAppMac and testpmd pods share the same name, so we can use it for doing
+# some extra checks
+    - name: check that at least the cnfappmac resource for the first testpmd pod has the required data
+      k8s_info:
+        namespace: metallb-system
```

Comment:
I wonder if this `namespace` and the `name` of the pod below need to be changed to match something in example-cnf?